// @mugan86/ng-shop-ui/esm5/lib/components/rating/rating.component.js
import { __decorate } from "tslib";
import { Component, Input } from '@angular/core';
var RatingComponent = /** @class */ (function () {
    function RatingComponent() {
        this.min = 1;
        this.max = 5;
        this.rating = undefined;
    }
    RatingComponent.prototype.ngOnInit = function () {
        if (this.rating === undefined || this.rating === null) {
            this.rating.value = 0;
            this.rating.count = 0;
        }
        this.starsCssValues = Array(this.max).fill('');
    };
    __decorate([
        Input()
    ], RatingComponent.prototype, "min", void 0);
    __decorate([
        Input()
    ], RatingComponent.prototype, "max", void 0);
    __decorate([
        Input()
    ], RatingComponent.prototype, "rating", void 0);
    RatingComponent = __decorate([
        Component({
            selector: 'shop-rating',
            template: "<div *ngIf=\"starsCssValues && rating.value\">\n  <span *ngFor=\"let v of starsCssValues; let i = index\">\n    <i class=\"fas fa-star\" [ngClass]=\"{ checked: rating.value >= i + 1 }\"></i>\n  </span>\n  {{ rating.value }} ({{ rating.count }})\n</div>\n",
            styles: ["\n      .checked {\n        color: orange;\n      }\n\n      .star,\n      .star-20,\n      .star-100 {\n        color: transparent;\n        -webkit-background-clip: text;\n        background-clip: text;\n      }\n\n      .star-20 {\n        background-image: -webkit-gradient(\n          linear,\n          left top,\n          right top,\n          color-stop(0.25, orange),\n          color-stop(1, black)\n        );\n        background-image: gradient(\n          linear,\n          left top,\n          right top,\n          color-stop(0.25, orange),\n          color-stop(1, black)\n        );\n      }\n\n      .star-100 {\n        background-image: -webkit-gradient(\n          linear,\n          left top,\n          right top,\n          color-stop(1, orange)\n        );\n        background-image: gradient(\n          linear,\n          left top,\n          right top,\n          color-stop(1, orange)\n        );\n      }\n    "]
        })
    ], RatingComponent);
    return RatingComponent;
}());
export { RatingComponent };
