// @mugan86/ng-shop-ui/esm5/lib/components/quantity-selector/quantity-selector.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
var QuantitySelectorComponent = /** @class */ (function () {
    function QuantitySelectorComponent() {
        this.updateValue = new EventEmitter();
    }
    QuantitySelectorComponent.prototype.changeValue = function (action) {
        if (action === '-' && this.qty > 1) {
            this.qty--;
        }
        else if (action === '+' && this.qty < this.stock) {
            this.qty++;
        }
        this.sendUpdate();
    };
    QuantitySelectorComponent.prototype.sendUpdate = function () {
        this.updateValue.emit(this.qty);
    };
    __decorate([
        Input()
    ], QuantitySelectorComponent.prototype, "qty", void 0);
    __decorate([
        Input()
    ], QuantitySelectorComponent.prototype, "stock", void 0);
    __decorate([
        Output()
    ], QuantitySelectorComponent.prototype, "updateValue", void 0);
    QuantitySelectorComponent = __decorate([
        Component({
            selector: 'shop-qty-selector',
            template: "<div>\n    <button type=\"button\" class=\"btn btn-dark qty-btn\" (click)=\"changeValue('-')\">\n        -\n      </button>\n      <input\n        type=\"text\"\n        [(ngModel)]=\"qty\"\n        value=\"1\"\n        name=\"quantity\"\n        class=\"text-center quantity\"\n        disabled\n      />\n      <button type=\"button\" class=\"btn btn-dark qty-btn\" (click)=\"changeValue('+')\">\n        +\n      </button>\n</div>\n\n",
            styles: ["\n      div {\n        margin-bottom: 5px;\n      }\n      .quantity {\n        width: 20%;\n        margin-bottom: 5px;\n      }\n\n      input[type='text']:disabled {\n        color: black;\n        width: 60px\n      }\n\n      .qty-btn {\n        border-radius: 20px;\n        padding-top: 3px;\n        /* top - right - bottom - left => sentido de las agujas del reloj*/\n        margin: -2px 3px 2px 3px;\n      }\n    "]
        })
    ], QuantitySelectorComponent);
    return QuantitySelectorComponent;
}());
export { QuantitySelectorComponent };
