// @mugan86/ng-shop-ui/esm2015/lib/components/quantity-selector/quantity-selector.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
let QuantitySelectorComponent = class QuantitySelectorComponent {
    constructor() {
        this.updateValue = new EventEmitter();
    }
    changeValue(action) {
        if (action === '-' && this.qty > 1) {
            this.qty--;
        }
        else if (action === '+' && this.qty < this.stock) {
            this.qty++;
        }
        this.sendUpdate();
    }
    sendUpdate() {
        this.updateValue.emit(this.qty);
    }
};
__decorate([
    Input()
], QuantitySelectorComponent.prototype, "qty", void 0);
__decorate([
    Input()
], QuantitySelectorComponent.prototype, "stock", void 0);
__decorate([
    Output()
], QuantitySelectorComponent.prototype, "updateValue", void 0);
QuantitySelectorComponent = __decorate([
    Component({
        selector: 'shop-qty-selector',
        template: "<div>\n    <button type=\"button\" class=\"btn btn-dark qty-btn\" (click)=\"changeValue('-')\">\n        -\n      </button>\n      <input\n        type=\"text\"\n        [(ngModel)]=\"qty\"\n        value=\"1\"\n        name=\"quantity\"\n        class=\"text-center quantity\"\n        disabled\n      />\n      <button type=\"button\" class=\"btn btn-dark qty-btn\" (click)=\"changeValue('+')\">\n        +\n      </button>\n</div>\n\n",
        styles: [`
      div {
        margin-bottom: 5px;
      }
      .quantity {
        width: 20%;
        margin-bottom: 5px;
      }

      input[type='text']:disabled {
        color: black;
        width: 60px
      }

      .qty-btn {
        border-radius: 20px;
        padding-top: 3px;
        /* top - right - bottom - left => sentido de las agujas del reloj*/
        margin: -2px 3px 2px 3px;
      }
    `]
    })
], QuantitySelectorComponent);
export { QuantitySelectorComponent };
