// @mugan86/ng-shop-ui/__ivy_ngcc__/fesm2015/mugan86-ng-shop-ui.js
import { __decorate } from 'tslib';
import { Input, Component, NgModule, EventEmitter, Output, Pipe } from '@angular/core';
import { CommonModule, SlicePipe } from '@angular/common';
import { Router } from '@angular/router';
import { FormsModule } from '@angular/forms';

import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '@angular/router';
import * as ɵngcc3 from '@angular/forms';

const _c0 = function (a0) { return { checked: a0 }; };
function RatingComponent_div_0_span_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵelement(1, "i", 2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const i_r3 = ctx.index;
    const ctx_r1 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(1, _c0, ctx_r1.rating.value >= i_r3 + 1));
} }
function RatingComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵtemplate(1, RatingComponent_div_0_span_1_Template, 2, 3, "span", 1);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r0.starsCssValues);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate2(" ", ctx_r0.rating.value, " (", ctx_r0.rating.count, ")\n");
} }
const _c1 = function (a0) { return { "active": a0 }; };
function CarouselItemsComponent_li_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "li", 10);
} if (rf & 2) {
    const slidePos_r3 = ctx.index;
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(1, _c1, slidePos_r3 == 0));
} }
const _c2 = function (a0) { return { "background-image": a0 }; };
function CarouselItemsComponent_div_5_Template(rf, ctx) { if (rf & 1) {
    const _r7 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 11);
    ɵngcc0.ɵɵlistener("click", function CarouselItemsComponent_div_5_Template_div_click_0_listener() { const restoredCtx = ɵngcc0.ɵɵrestoreView(_r7); const i_r5 = restoredCtx.index; const ctx_r6 = ɵngcc0.ɵɵnextContext(); return ctx_r6.open(i_r5); });
    ɵngcc0.ɵɵelementStart(1, "div", 12);
    ɵngcc0.ɵɵelementStart(2, "h3");
    ɵngcc0.ɵɵtext(3);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "p");
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r4 = ctx.$implicit;
    const i_r5 = ctx.index;
    ɵngcc0.ɵɵpropertyInterpolate2("title", "", item_r4.title, " - ", item_r4.description, "");
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(6, _c1, i_r5 == 0))("ngStyle", ɵngcc0.ɵɵpureFunction1(8, _c2, "url(" + item_r4.background + ")"));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate(item_r4.title);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(item_r4.description);
} }
function ProductItemComponent_p_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "p");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "truncate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind2(2, 1, ctx_r0.product.description, 40), " ");
} }
function ProductItemComponent_ng_template_10_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 15);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "number");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate2(" ", ctx_r2.selectCurrency, " ", ɵngcc0.ɵɵpipeBind1(2, 2, ctx_r2.product.price), " ");
} }
function ProductItemComponent_div_12_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 15);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "number");
    ɵngcc0.ɵɵelementStart(3, "span");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵpipe(5, "number");
    ɵngcc0.ɵɵelementStart(6, "span", 16);
    ɵngcc0.ɵɵtext(7);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate2(" ", ctx_r3.selectCurrency, " ", ɵngcc0.ɵɵpipeBind1(2, 5, ctx_r3.product.priceDiscount), " ");
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate2(" ", ctx_r3.selectCurrency, " ", ɵngcc0.ɵɵpipeBind1(5, 7, ctx_r3.product.price), " ");
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1("(", ctx_r3.product.discount, "%)");
} }
function ProductItemComponent_shop_rating_13_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "shop-rating", 17);
} if (rf & 2) {
    const ctx_r4 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("rating", ctx_r4.product.rating);
} }
var CURRENCY_LIST;
(function (CURRENCY_LIST) {
    CURRENCY_LIST["EURO"] = "EUR";
    CURRENCY_LIST["US_DOLLAR"] = "USD";
    CURRENCY_LIST["BRITISH_POUND"] = "GBP";
    CURRENCY_LIST["CANADIAN_DOLAR"] = "CAD";
    CURRENCY_LIST["COLOMBIAN_PESO"] = "COP";
    CURRENCY_LIST["CHILEAN_PESO"] = "CLP";
    CURRENCY_LIST["MEXICAN_PESO"] = "MXN";
    CURRENCY_LIST["VENEZUELA_BOLIVAR"] = "VSD";
    CURRENCY_LIST["PERUVIAN_SOL"] = "PEN";
    CURRENCY_LIST["ARGENTINA_PESO"] = "ARS";
})(CURRENCY_LIST || (CURRENCY_LIST = {}));
var CURRENCIES_SYMBOL;
(function (CURRENCIES_SYMBOL) {
    CURRENCIES_SYMBOL["EUR"] = "\u20AC";
    CURRENCIES_SYMBOL["USD"] = "$";
    CURRENCIES_SYMBOL["GBP"] = "\u00A3";
    CURRENCIES_SYMBOL["CAD"] = "$";
    CURRENCIES_SYMBOL["COP"] = "$";
    CURRENCIES_SYMBOL["CLP"] = "$";
    CURRENCIES_SYMBOL["MXN"] = "$";
    CURRENCIES_SYMBOL["VSD"] = "Bs.";
    CURRENCIES_SYMBOL["PEN"] = "S/.";
    CURRENCIES_SYMBOL["ARS"] = "$";
})(CURRENCIES_SYMBOL || (CURRENCIES_SYMBOL = {}));

let RatingComponent = class RatingComponent {
    constructor() {
        this.min = 1;
        this.max = 5;
        this.rating = undefined;
    }
    ngOnInit() {
        if (this.rating === undefined || this.rating === null) {
            this.rating.value = 0;
            this.rating.count = 0;
        }
        this.starsCssValues = Array(this.max).fill('');
    }
};
RatingComponent.ɵfac = function RatingComponent_Factory(t) { return new (t || RatingComponent)(); };
RatingComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: RatingComponent, selectors: [["shop-rating"]], inputs: { min: "min", max: "max", rating: "rating" }, decls: 1, vars: 1, consts: [[4, "ngIf"], [4, "ngFor", "ngForOf"], [1, "fas", "fa-star", 3, "ngClass"]], template: function RatingComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, RatingComponent_div_0_Template, 3, 3, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.starsCssValues && ctx.rating.value);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgForOf, ɵngcc1.NgClass], styles: [".checked[_ngcontent-%COMP%] {\n        color: orange;\n      }\n\n      .star[_ngcontent-%COMP%], .star-20[_ngcontent-%COMP%], .star-100[_ngcontent-%COMP%] {\n        color: transparent;\n        -webkit-background-clip: text;\n        background-clip: text;\n      }\n\n      .star-20[_ngcontent-%COMP%] {\n        background-image: -webkit-gradient(\n          linear,\n          left top,\n          right top,\n          color-stop(0.25, orange),\n          color-stop(1, black)\n        );\n        background-image: gradient(\n          linear,\n          left top,\n          right top,\n          color-stop(0.25, orange),\n          color-stop(1, black)\n        );\n      }\n\n      .star-100[_ngcontent-%COMP%] {\n        background-image: -webkit-gradient(\n          linear,\n          left top,\n          right top,\n          color-stop(1, orange)\n        );\n        background-image: gradient(\n          linear,\n          left top,\n          right top,\n          color-stop(1, orange)\n        );\n      }"] });
__decorate([
    Input()
], RatingComponent.prototype, "min", void 0);
__decorate([
    Input()
], RatingComponent.prototype, "max", void 0);
__decorate([
    Input()
], RatingComponent.prototype, "rating", void 0);

let RatingModule = class RatingModule {
};
RatingModule.ɵfac = function RatingModule_Factory(t) { return new (t || RatingModule)(); };
RatingModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: RatingModule });
RatingModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });

let CarouselItemsComponent = class CarouselItemsComponent {
    constructor(router) {
        this.router = router;
        this.items = [];
    }
    open(i) {
        const urlsCommands = this.items[i].url.split('/');
        this.router.navigate([...urlsCommands]);
    }
};
CarouselItemsComponent.ɵfac = function CarouselItemsComponent_Factory(t) { return new (t || CarouselItemsComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc2.Router)); };
CarouselItemsComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: CarouselItemsComponent, selectors: [["shop-carousel-items"]], inputs: { items: "items" }, decls: 14, vars: 2, consts: [["id", "carouselExampleIndicators", "data-ride", "carousel", 1, "carousel", "slide"], [1, "carousel-indicators"], ["data-target", "#carouselExampleIndicators", "data-slide-to", "slidePos", 3, "ngClass", 4, "ngFor", "ngForOf"], ["role", "listbox", 1, "carousel-inner"], ["class", "carousel-item url-item", 3, "title", "ngClass", "ngStyle", "click", 4, "ngFor", "ngForOf"], ["href", "#carouselExampleIndicators", "role", "button", "data-slide", "prev", 1, "carousel-control-prev"], ["aria-hidden", "true", 1, "carousel-control-prev-icon"], [1, "sr-only"], ["href", "#carouselExampleIndicators", "role", "button", "data-slide", "next", 1, "carousel-control-next"], ["aria-hidden", "true", 1, "carousel-control-next-icon"], ["data-target", "#carouselExampleIndicators", "data-slide-to", "slidePos", 3, "ngClass"], [1, "carousel-item", "url-item", 3, "title", "ngClass", "ngStyle", "click"], [1, "carousel-caption", "d-none", "d-md-block"]], template: function CarouselItemsComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "section");
        ɵngcc0.ɵɵelementStart(1, "div", 0);
        ɵngcc0.ɵɵelementStart(2, "ol", 1);
        ɵngcc0.ɵɵtemplate(3, CarouselItemsComponent_li_3_Template, 1, 3, "li", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "div", 3);
        ɵngcc0.ɵɵtemplate(5, CarouselItemsComponent_div_5_Template, 6, 10, "div", 4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(6, "a", 5);
        ɵngcc0.ɵɵelement(7, "span", 6);
        ɵngcc0.ɵɵelementStart(8, "span", 7);
        ɵngcc0.ɵɵtext(9, "Previous");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(10, "a", 8);
        ɵngcc0.ɵɵelement(11, "span", 9);
        ɵngcc0.ɵɵelementStart(12, "span", 7);
        ɵngcc0.ɵɵtext(13, "Next");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.items);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.items);
    } }, directives: [ɵngcc1.NgForOf, ɵngcc1.NgClass, ɵngcc1.NgStyle], styles: ["section[_ngcontent-%COMP%] {\n        margin-bottom: 25px\n      }\n      .url-item[_ngcontent-%COMP%] {\n        cursor: pointer;\n      }\n\n      .carousel-inner[_ngcontent-%COMP%] {\n        height: 500px;\n      }\n\n      .carousel-item[_ngcontent-%COMP%] {\n        height: 100%;\n        min-height: 350px;\n        background: no-repeat center center scroll;\n        -webkit-background-size: cover;\n        -moz-background-size: cover;\n        -o-background-size: cover;\n        background-size: cover;\n      }\n\n      @media (max-width: 1440px) {\n        .carousel-item[_ngcontent-%COMP%] {\n          max-height: 640px;\n        }\n      }\n      @media (max-width: 400px) {\n        .carousel-item[_ngcontent-%COMP%] {\n          max-height: 240px;\n        }\n      }"] });
CarouselItemsComponent.ctorParameters = () => [
    { type: Router }
];
__decorate([
    Input()
], CarouselItemsComponent.prototype, "items", void 0);

let CarouselItemsModule = class CarouselItemsModule {
};
CarouselItemsModule.ɵfac = function CarouselItemsModule_Factory(t) { return new (t || CarouselItemsModule)(); };
CarouselItemsModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: CarouselItemsModule });
CarouselItemsModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });

let ProductItemComponent = class ProductItemComponent {
    constructor() {
        this.showDesc = false;
        this.add = new EventEmitter();
        this.itemDetails = new EventEmitter();
        this.selectCurrency = CURRENCIES_SYMBOL[CURRENCY_LIST.EURO];
    }
    ngOnInit() {
        this.product.qty = 0;
    }
    addToCart(product) {
        product.qty = product.qty + 1;
        this.add.emit(product);
    }
    showDetails(product) {
        this.itemDetails.emit(product);
    }
};
ProductItemComponent.ɵfac = function ProductItemComponent_Factory(t) { return new (t || ProductItemComponent)(); };
ProductItemComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: ProductItemComponent, selectors: [["shop-product-item"]], inputs: { showDesc: "showDesc", selectCurrency: "selectCurrency", product: "product" }, outputs: { add: "add", itemDetails: "itemDetails" }, decls: 21, vars: 10, consts: [[1, "product-grid"], [1, "product-image"], [3, "click"], [1, "pic-1", 3, "src", "alt"], [1, "product-content"], [1, "title"], [4, "ngIf"], ["noDiscount", ""], ["class", "price", 4, "ngIf", "ngIfElse"], [3, "rating", 4, "ngIf"], [1, "social"], ["data-tip", "A\u00F1adir a la cesta", 1, "item-button-options", 3, "click"], [1, "fas", "fa-cart-plus"], ["data-tip", "Detalles del producto", 1, "item-button-options", 3, "click"], [1, "fas", "fa-info"], [1, "price"], [1, "discount-percentage"], [3, "rating"]], template: function ProductItemComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelementStart(2, "a", 2);
        ɵngcc0.ɵɵlistener("click", function ProductItemComponent_Template_a_click_2_listener() { return ctx.showDetails(ctx.product); });
        ɵngcc0.ɵɵelement(3, "img", 3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "div", 4);
        ɵngcc0.ɵɵelementStart(5, "h3", 5);
        ɵngcc0.ɵɵtext(6);
        ɵngcc0.ɵɵpipe(7, "truncate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(8, ProductItemComponent_p_8_Template, 3, 4, "p", 6);
        ɵngcc0.ɵɵelementStart(9, "div");
        ɵngcc0.ɵɵtemplate(10, ProductItemComponent_ng_template_10_Template, 3, 4, "ng-template", null, 7, ɵngcc0.ɵɵtemplateRefExtractor);
        ɵngcc0.ɵɵtemplate(12, ProductItemComponent_div_12_Template, 8, 9, "div", 8);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(13, ProductItemComponent_shop_rating_13_Template, 1, 1, "shop-rating", 9);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(14, "ul", 10);
        ɵngcc0.ɵɵelementStart(15, "li");
        ɵngcc0.ɵɵelementStart(16, "a", 11);
        ɵngcc0.ɵɵlistener("click", function ProductItemComponent_Template_a_click_16_listener() { return ctx.addToCart(ctx.product); });
        ɵngcc0.ɵɵelement(17, "i", 12);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(18, "li");
        ɵngcc0.ɵɵelementStart(19, "a", 13);
        ɵngcc0.ɵɵlistener("click", function ProductItemComponent_Template_a_click_19_listener() { return ctx.showDetails(ctx.product); });
        ɵngcc0.ɵɵelement(20, "i", 14);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        const _r1 = ɵngcc0.ɵɵreference(11);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵpropertyInterpolate("src", ctx.product.img, ɵngcc0.ɵɵsanitizeUrl);
        ɵngcc0.ɵɵproperty("alt", ctx.product.name);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind2(7, 7, ctx.product.name, 25));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showDesc);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵproperty("ngIf", ctx.product.discount)("ngIfElse", _r1);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.product.rating);
    } }, directives: function () { return [ɵngcc1.NgIf, RatingComponent]; }, pipes: function () { return [TruncatePipe, ɵngcc1.DecimalPipe]; }, styles: [".product-grid[_ngcontent-%COMP%], .product-grid[_ngcontent-%COMP%]   .product-image[_ngcontent-%COMP%] {\n        overflow: hidden;\n      }\n      .product-grid[_ngcontent-%COMP%] {\n        font-family: 'Open Sans', sans-serif;\n        text-align: center;\n        position: relative;\n        transition: all 0.5s ease 0s;\n        border: 1px solid #afa3a338;\n        border-radius: 8px;\n      }\n      .product-grid[_ngcontent-%COMP%]:hover {\n        box-shadow: 0 0 10px rgba(0, 0, 0, 0.3);\n        cursor: pointer\n      }\n      .product-grid[_ngcontent-%COMP%]   .product-image[_ngcontent-%COMP%]   a[_ngcontent-%COMP%] {\n        display: block;\n      }\n      .product-grid[_ngcontent-%COMP%]   .product-image[_ngcontent-%COMP%]   img[_ngcontent-%COMP%] {\n        width: 100%;\n        min-height: 150px;\n        max-height: 150px;\n        transition: all 0.5s ease 0s;\n      }\n      .product-grid[_ngcontent-%COMP%]:hover   .product-image[_ngcontent-%COMP%]   img[_ngcontent-%COMP%] {\n        transform: scale(1.1);\n      }\n      .product-grid[_ngcontent-%COMP%]   .product-content[_ngcontent-%COMP%] {\n        padding: 12px 12px 15px;\n        transition: all 0.5s ease 0s;\n      }\n      .product-grid[_ngcontent-%COMP%]:hover   .product-content[_ngcontent-%COMP%] {\n        opacity: 0;\n      }\n      .product-grid[_ngcontent-%COMP%]   .title[_ngcontent-%COMP%] {\n        font-size: 18px;\n        font-weight: 600;\n        text-transform: capitalize;\n        margin: 0 0 10px;\n        transition: all 0.3s ease 0s;\n      }\n      .product-grid[_ngcontent-%COMP%]   .title[_ngcontent-%COMP%]   a[_ngcontent-%COMP%] {\n        color: #000;\n      }\n      .product-grid[_ngcontent-%COMP%]   .title[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]:hover {\n        color: #333;\n      }\n      .product-grid[_ngcontent-%COMP%]   .price[_ngcontent-%COMP%] {\n        font-size: 22px;\n        font-weight: 800;\n        color: #cc1c39;\n        text-align: left;\n        margin-left: 20px;\n        margin-bottom: 12px;\n      }\n      .product-grid[_ngcontent-%COMP%]   .price[_ngcontent-%COMP%]   span[_ngcontent-%COMP%] {\n        color: #999;\n        font-size: 15px;\n        font-weight: 400;\n        text-decoration: line-through;\n        margin-left: 7px;\n        display: inline-block;\n      }\n      .product-grid[_ngcontent-%COMP%]   .price[_ngcontent-%COMP%]   span.discount-percentage[_ngcontent-%COMP%] {\n        text-decoration: none;\n      }\n      .product-grid[_ngcontent-%COMP%]   .social[_ngcontent-%COMP%] {\n        background-color: #fff;\n        width: 100%;\n        padding: 0;\n        margin: 0;\n        list-style: none;\n        opacity: 0;\n        transform: translateX(-50%);\n        position: absolute;\n        bottom: -50%;\n        left: 50%;\n        z-index: 1;\n        transition: all 0.5s ease 0s;\n      }\n      .product-grid[_ngcontent-%COMP%]:hover   .social[_ngcontent-%COMP%] {\n        opacity: 1;\n        bottom: 20px;\n      }\n\n      .product-grid[_ngcontent-%COMP%]   .social[_ngcontent-%COMP%]   li[_ngcontent-%COMP%] {\n        display: inline-block;\n      }\n      .product-grid[_ngcontent-%COMP%]   .social[_ngcontent-%COMP%]   li[_ngcontent-%COMP%]   a[_ngcontent-%COMP%] {\n        color: #909090;\n        font-size: 16px;\n        line-height: 45px;\n        text-align: center;\n        height: 45px;\n        width: 45px;\n        margin: 0 7px;\n        border: 1px solid #909090;\n        border-radius: 50px;\n        display: block;\n        position: relative;\n        transition: all 0.3s ease-in-out;\n      }\n      .product-grid[_ngcontent-%COMP%]   .social[_ngcontent-%COMP%]   li[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]:hover {\n        color: #fff;\n        background-color: #333;\n        width: 80px;\n      }\n      .product-grid[_ngcontent-%COMP%]   .social[_ngcontent-%COMP%]   li[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]:after, .product-grid[_ngcontent-%COMP%]   .social[_ngcontent-%COMP%]   li[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]:before {\n        content: attr(data-tip);\n        color: #fff;\n        background-color: #333;\n        font-size: 12px;\n        letter-spacing: 1px;\n        line-height: 20px;\n        padding: 1px 5px;\n        border-radius: 5px;\n        white-space: nowrap;\n        opacity: 0;\n        transform: translateX(-50%);\n        position: absolute;\n        left: 50%;\n        top: -30px;\n      }\n      .product-grid[_ngcontent-%COMP%]   .social[_ngcontent-%COMP%]   li[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]:after {\n        content: '';\n        height: 15px;\n        width: 15px;\n        border-radius: 0;\n        transform: translateX(-50%) rotate(45deg);\n        top: -20px;\n        z-index: -1;\n      }\n      .product-grid[_ngcontent-%COMP%]   .social[_ngcontent-%COMP%]   li[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]:hover:after, .product-grid[_ngcontent-%COMP%]   .social[_ngcontent-%COMP%]   li[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]:hover:before {\n        opacity: 1;\n      }\n      .item-button-options[_ngcontent-%COMP%]:hover {\n        background: #504a4a;\n        cursor: pointer;\n      }\n      @media only screen and (max-width: 990px) {\n        .product-grid[_ngcontent-%COMP%] {\n          margin-bottom: 30px;\n        }\n      }"] });
__decorate([
    Input()
], ProductItemComponent.prototype, "product", void 0);
__decorate([
    Input()
], ProductItemComponent.prototype, "showDesc", void 0);
__decorate([
    Output()
], ProductItemComponent.prototype, "add", void 0);
__decorate([
    Output()
], ProductItemComponent.prototype, "itemDetails", void 0);
__decorate([
    Input()
], ProductItemComponent.prototype, "selectCurrency", void 0);

let TruncatePipe = class TruncatePipe {
    transform(value, limit = 25) {
        return value.length > limit
            ? new SlicePipe().transform(value, 0, limit - 3).concat('...')
            : value;
    }
};
TruncatePipe.ɵfac = function TruncatePipe_Factory(t) { return new (t || TruncatePipe)(); };
TruncatePipe.ɵpipe = /*@__PURE__*/ ɵngcc0.ɵɵdefinePipe({ name: "truncate", type: TruncatePipe, pure: true });

let ProductItemModule = class ProductItemModule {
};
ProductItemModule.ɵfac = function ProductItemModule_Factory(t) { return new (t || ProductItemModule)(); };
ProductItemModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: ProductItemModule });
ProductItemModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule,
            RatingModule
        ]] });

const MODULES = [
    FormsModule,
    RatingModule,
    CarouselItemsModule,
    ProductItemModule
];
let ShopUiModule = class ShopUiModule {
};
ShopUiModule.ɵfac = function ShopUiModule_Factory(t) { return new (t || ShopUiModule)(); };
ShopUiModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: ShopUiModule });
ShopUiModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [MODULES, FormsModule, RatingModule, CarouselItemsModule, ProductItemModule] });

let QuantitySelectorComponent = class QuantitySelectorComponent {
    constructor() {
        this.updateValue = new EventEmitter();
    }
    changeValue(action) {
        if (action === '-' && this.qty > 1) {
            this.qty--;
        }
        else if (action === '+' && this.qty < this.stock) {
            this.qty++;
        }
        this.sendUpdate();
    }
    sendUpdate() {
        this.updateValue.emit(this.qty);
    }
};
QuantitySelectorComponent.ɵfac = function QuantitySelectorComponent_Factory(t) { return new (t || QuantitySelectorComponent)(); };
QuantitySelectorComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: QuantitySelectorComponent, selectors: [["shop-qty-selector"]], inputs: { qty: "qty", stock: "stock" }, outputs: { updateValue: "updateValue" }, decls: 6, vars: 1, consts: [["type", "button", 1, "btn", "btn-dark", "qty-btn", 3, "click"], ["type", "text", "value", "1", "name", "quantity", "disabled", "", 1, "text-center", "quantity", 3, "ngModel", "ngModelChange"]], template: function QuantitySelectorComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵelementStart(1, "button", 0);
        ɵngcc0.ɵɵlistener("click", function QuantitySelectorComponent_Template_button_click_1_listener() { return ctx.changeValue("-"); });
        ɵngcc0.ɵɵtext(2, " - ");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "input", 1);
        ɵngcc0.ɵɵlistener("ngModelChange", function QuantitySelectorComponent_Template_input_ngModelChange_3_listener($event) { return ctx.qty = $event; });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "button", 0);
        ɵngcc0.ɵɵlistener("click", function QuantitySelectorComponent_Template_button_click_4_listener() { return ctx.changeValue("+"); });
        ɵngcc0.ɵɵtext(5, " + ");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngModel", ctx.qty);
    } }, directives: [ɵngcc3.DefaultValueAccessor, ɵngcc3.NgControlStatus, ɵngcc3.NgModel], styles: ["div[_ngcontent-%COMP%] {\n        margin-bottom: 5px;\n      }\n      .quantity[_ngcontent-%COMP%] {\n        width: 20%;\n        margin-bottom: 5px;\n      }\n\n      input[type='text'][_ngcontent-%COMP%]:disabled {\n        color: black;\n        width: 60px\n      }\n\n      .qty-btn[_ngcontent-%COMP%] {\n        border-radius: 20px;\n        padding-top: 3px;\n        \n        margin: -2px 3px 2px 3px;\n      }"] });
__decorate([
    Input()
], QuantitySelectorComponent.prototype, "qty", void 0);
__decorate([
    Input()
], QuantitySelectorComponent.prototype, "stock", void 0);
__decorate([
    Output()
], QuantitySelectorComponent.prototype, "updateValue", void 0);

let QuantitySelectorModule = class QuantitySelectorModule {
};
QuantitySelectorModule.ɵfac = function QuantitySelectorModule_Factory(t) { return new (t || QuantitySelectorModule)(); };
QuantitySelectorModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: QuantitySelectorModule });
QuantitySelectorModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule,
            FormsModule
        ]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(RatingComponent, [{
        type: Component,
        args: [{
                selector: 'shop-rating',
                template: "<div *ngIf=\"starsCssValues && rating.value\">\n  <span *ngFor=\"let v of starsCssValues; let i = index\">\n    <i class=\"fas fa-star\" [ngClass]=\"{ checked: rating.value >= i + 1 }\"></i>\n  </span>\n  {{ rating.value }} ({{ rating.count }})\n</div>\n",
                styles: [`
      .checked {
        color: orange;
      }

      .star,
      .star-20,
      .star-100 {
        color: transparent;
        -webkit-background-clip: text;
        background-clip: text;
      }

      .star-20 {
        background-image: -webkit-gradient(
          linear,
          left top,
          right top,
          color-stop(0.25, orange),
          color-stop(1, black)
        );
        background-image: gradient(
          linear,
          left top,
          right top,
          color-stop(0.25, orange),
          color-stop(1, black)
        );
      }

      .star-100 {
        background-image: -webkit-gradient(
          linear,
          left top,
          right top,
          color-stop(1, orange)
        );
        background-image: gradient(
          linear,
          left top,
          right top,
          color-stop(1, orange)
        );
      }
    `]
            }]
    }], function () { return []; }, { min: [{
            type: Input
        }], max: [{
            type: Input
        }], rating: [{
            type: Input
        }] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(RatingModule, [{
        type: NgModule,
        args: [{
                declarations: [RatingComponent],
                imports: [
                    CommonModule
                ],
                exports: [RatingComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(RatingModule, { declarations: function () { return [RatingComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [RatingComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(CarouselItemsComponent, [{
        type: Component,
        args: [{
                selector: 'shop-carousel-items',
                template: "<section>\n    <div id=\"carouselExampleIndicators\" class=\"carousel slide\" data-ride=\"carousel\">\n      <ol class=\"carousel-indicators\">\n        <li data-target=\"#carouselExampleIndicators\" *ngFor=\"let slide of items; let slidePos = index\" data-slide-to=\"slidePos\" [ngClass]=\"{'active' : slidePos == 0}\"></li>\n      </ol>\n      <div class=\"carousel-inner\" role=\"listbox\">\n        <div *ngFor=\"let item of items; let i = index\" class=\"carousel-item url-item\" \n        title=\"{{ item.title }} - {{ item.description }}\"\n        [ngClass]=\"{'active' : i == 0}\" (click)=\"open(i)\"\n        [ngStyle]=\"{'background-image': 'url(' + item.background + ')'}\">\n          <div class=\"carousel-caption d-none d-md-block\">\n            <h3>{{ item.title }}</h3>\n            <p>{{ item.description }}</p>\n          </div>\n        </div>\n      </div>\n      <a class=\"carousel-control-prev\" href=\"#carouselExampleIndicators\" role=\"button\" data-slide=\"prev\">\n            <span class=\"carousel-control-prev-icon\" aria-hidden=\"true\"></span>\n            <span class=\"sr-only\">Previous</span>\n          </a>\n      <a class=\"carousel-control-next\" href=\"#carouselExampleIndicators\" role=\"button\" data-slide=\"next\">\n            <span class=\"carousel-control-next-icon\" aria-hidden=\"true\"></span>\n            <span class=\"sr-only\">Next</span>\n          </a>\n    </div>\n  </section>\n  ",
                styles: [`
      section {
        margin-bottom: 25px
      }
      .url-item {
        cursor: pointer;
      }

      .carousel-inner {
        height: 500px;
      }

      .carousel-item {
        height: 100%;
        min-height: 350px;
        background: no-repeat center center scroll;
        -webkit-background-size: cover;
        -moz-background-size: cover;
        -o-background-size: cover;
        background-size: cover;
      }

      @media (max-width: 1440px) {
        .carousel-item {
          max-height: 640px;
        }
      }
      @media (max-width: 400px) {
        .carousel-item {
          max-height: 240px;
        }
      }
    `]
            }]
    }], function () { return [{ type: ɵngcc2.Router }]; }, { items: [{
            type: Input
        }] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(CarouselItemsModule, [{
        type: NgModule,
        args: [{
                declarations: [CarouselItemsComponent],
                imports: [
                    CommonModule
                ],
                exports: [CarouselItemsComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CarouselItemsModule, { declarations: function () { return [CarouselItemsComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [CarouselItemsComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ProductItemComponent, [{
        type: Component,
        args: [{
                selector: 'shop-product-item',
                template: "<div class=\"product-grid\">\n  <div class=\"product-image\">\n    <a (click)=\"showDetails(product)\">\n      <img src=\"{{ product.img }}\" class=\"pic-1\" [alt]=\"product.name\" />\n    </a>\n  </div>\n  <div class=\"product-content\">\n    <h3 class=\"title\">{{ product.name | truncate: 25 }}</h3>\n    <p *ngIf=\"showDesc\">\n      {{ product.description | truncate: 40 }}\n    </p>\n    <div>\n      <ng-template #noDiscount>\n        <div class=\"price\">\n          {{ selectCurrency }} {{ product.price | number }}\n        </div>\n      </ng-template>\n      <div class=\"price\" *ngIf=\"product.discount; else noDiscount\">\n        {{ selectCurrency }} {{ product.priceDiscount | number }}\n        <span >\n          {{ selectCurrency }}  {{ product.price | number }}\n          <span class=\"discount-percentage\">({{ product.discount }}%)</span>\n        </span>\n      </div>\n    </div>\n    <shop-rating *ngIf=\"product.rating\" [rating]=\"product.rating\"></shop-rating>\n  </div>\n  <ul class=\"social\">\n    <!--<li><a href=\"\" data-tip=\"A\u00F1adir a lista de deseos\"><i class=\"fa fa-shopping-bag\"></i></a></li>-->\n    <li>\n      <a\n        class=\"item-button-options\"\n        (click)=\"addToCart(product)\"\n        data-tip=\"A\u00F1adir a la cesta\"\n        ><i class=\"fas fa-cart-plus\"></i\n      ></a>\n    </li>\n    <li>\n      <a class=\"item-button-options\"\n        (click)=\"showDetails(product)\" data-tip=\"Detalles del producto\"\n        ><i class=\"fas fa-info\"></i\n      ></a>\n    </li>\n  </ul>\n</div>\n",
                styles: [`
      /********************* Shopping Demo-6 **********************/
      .product-grid,
      .product-grid .product-image {
        overflow: hidden;
      }
      .product-grid {
        font-family: 'Open Sans', sans-serif;
        text-align: center;
        position: relative;
        transition: all 0.5s ease 0s;
        border: 1px solid #afa3a338;
        border-radius: 8px;
      }
      .product-grid:hover {
        box-shadow: 0 0 10px rgba(0, 0, 0, 0.3);
        cursor: pointer
      }
      .product-grid .product-image a {
        display: block;
      }
      .product-grid .product-image img {
        width: 100%;
        min-height: 150px;
        max-height: 150px;
        transition: all 0.5s ease 0s;
      }
      .product-grid:hover .product-image img {
        transform: scale(1.1);
      }
      .product-grid .product-content {
        padding: 12px 12px 15px;
        transition: all 0.5s ease 0s;
      }
      .product-grid:hover .product-content {
        opacity: 0;
      }
      .product-grid .title {
        font-size: 18px;
        font-weight: 600;
        text-transform: capitalize;
        margin: 0 0 10px;
        transition: all 0.3s ease 0s;
      }
      .product-grid .title a {
        color: #000;
      }
      .product-grid .title a:hover {
        color: #333;
      }
      .product-grid .price {
        font-size: 22px;
        font-weight: 800;
        color: #cc1c39;
        text-align: left;
        margin-left: 20px;
        margin-bottom: 12px;
      }
      .product-grid .price span {
        color: #999;
        font-size: 15px;
        font-weight: 400;
        text-decoration: line-through;
        margin-left: 7px;
        display: inline-block;
      }
      .product-grid .price span.discount-percentage {
        text-decoration: none;
      }
      .product-grid .social {
        background-color: #fff;
        width: 100%;
        padding: 0;
        margin: 0;
        list-style: none;
        opacity: 0;
        transform: translateX(-50%);
        position: absolute;
        bottom: -50%;
        left: 50%;
        z-index: 1;
        transition: all 0.5s ease 0s;
      }
      .product-grid:hover .social {
        opacity: 1;
        bottom: 20px;
      }

      .product-grid .social li {
        display: inline-block;
      }
      .product-grid .social li a {
        color: #909090;
        font-size: 16px;
        line-height: 45px;
        text-align: center;
        height: 45px;
        width: 45px;
        margin: 0 7px;
        border: 1px solid #909090;
        border-radius: 50px;
        display: block;
        position: relative;
        transition: all 0.3s ease-in-out;
      }
      .product-grid .social li a:hover {
        color: #fff;
        background-color: #333;
        width: 80px;
      }
      .product-grid .social li a:after,
      .product-grid .social li a:before {
        content: attr(data-tip);
        color: #fff;
        background-color: #333;
        font-size: 12px;
        letter-spacing: 1px;
        line-height: 20px;
        padding: 1px 5px;
        border-radius: 5px;
        white-space: nowrap;
        opacity: 0;
        transform: translateX(-50%);
        position: absolute;
        left: 50%;
        top: -30px;
      }
      .product-grid .social li a:after {
        content: '';
        height: 15px;
        width: 15px;
        border-radius: 0;
        transform: translateX(-50%) rotate(45deg);
        top: -20px;
        z-index: -1;
      }
      .product-grid .social li a:hover:after,
      .product-grid .social li a:hover:before {
        opacity: 1;
      }
      .item-button-options:hover {
        background: #504a4a;
        cursor: pointer;
      }
      @media only screen and (max-width: 990px) {
        .product-grid {
          margin-bottom: 30px;
        }
      }
    `]
            }]
    }], function () { return []; }, { showDesc: [{
            type: Input
        }], add: [{
            type: Output
        }], itemDetails: [{
            type: Output
        }], selectCurrency: [{
            type: Input
        }], product: [{
            type: Input
        }] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(TruncatePipe, [{
        type: Pipe,
        args: [{
                name: 'truncate'
            }]
    }], null, null); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ProductItemModule, [{
        type: NgModule,
        args: [{
                declarations: [ProductItemComponent, TruncatePipe],
                imports: [
                    CommonModule,
                    RatingModule
                ],
                exports: [ProductItemComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ProductItemModule, { declarations: function () { return [ProductItemComponent, TruncatePipe]; }, imports: function () { return [CommonModule, RatingModule]; }, exports: function () { return [ProductItemComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ShopUiModule, [{
        type: NgModule,
        args: [{
                imports: MODULES,
                exports: MODULES
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ShopUiModule, { imports: [FormsModule, RatingModule, CarouselItemsModule, ProductItemModule], exports: [FormsModule, RatingModule, CarouselItemsModule, ProductItemModule] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(QuantitySelectorComponent, [{
        type: Component,
        args: [{
                selector: 'shop-qty-selector',
                template: "<div>\n    <button type=\"button\" class=\"btn btn-dark qty-btn\" (click)=\"changeValue('-')\">\n        -\n      </button>\n      <input\n        type=\"text\"\n        [(ngModel)]=\"qty\"\n        value=\"1\"\n        name=\"quantity\"\n        class=\"text-center quantity\"\n        disabled\n      />\n      <button type=\"button\" class=\"btn btn-dark qty-btn\" (click)=\"changeValue('+')\">\n        +\n      </button>\n</div>\n\n",
                styles: [`
      div {
        margin-bottom: 5px;
      }
      .quantity {
        width: 20%;
        margin-bottom: 5px;
      }

      input[type='text']:disabled {
        color: black;
        width: 60px
      }

      .qty-btn {
        border-radius: 20px;
        padding-top: 3px;
        /* top - right - bottom - left => sentido de las agujas del reloj*/
        margin: -2px 3px 2px 3px;
      }
    `]
            }]
    }], function () { return []; }, { updateValue: [{
            type: Output
        }], qty: [{
            type: Input
        }], stock: [{
            type: Input
        }] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(QuantitySelectorModule, [{
        type: NgModule,
        args: [{
                declarations: [QuantitySelectorComponent],
                imports: [
                    CommonModule,
                    FormsModule
                ],
                exports: [QuantitySelectorComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(QuantitySelectorModule, { declarations: function () { return [QuantitySelectorComponent]; }, imports: function () { return [CommonModule,
        FormsModule]; }, exports: function () { return [QuantitySelectorComponent]; } }); })();

/*
 * Public API Surface of shop-ui
 */

/**
 * Generated bundle index. Do not edit.
 */

export { CURRENCIES_SYMBOL, CURRENCY_LIST, CarouselItemsComponent, CarouselItemsModule, ProductItemComponent, ProductItemModule, QuantitySelectorComponent, QuantitySelectorModule, RatingComponent, RatingModule, ShopUiModule, TruncatePipe };
